pyembed/core/discovery.py

```diff
@@ -219,12 +219,16 @@ def __init__(self, url):
     def get_oembed_urls(self, url, oembed_format=None):
         if not self.endpoints:
             response = requests.get(self.url)
+            try:
+                providers = response.json()
+            except ValueError:
+                providers = None
```

Comment:
Think I'd prefer to raise a `PyEmbedDiscoveryError` here, rather than allocating None to providers - this means that you could avoid the `or not providers` check below.